dataplane/linux/int_dataplane.go

```diff
@@ -1077,6 +1081,7 @@ func (d *InternalDataplane) apply() {
 		ipSetsWG.Add(1)
 		go func(ipSets *ipsets.IPSets) {
 			ipSets.ApplyUpdates()
+			d.reportHealth()
```

Comment:
Will this be thread-safe?
At this level:
```
	if d.config.HealthAggregator != nil {
		d.config.HealthAggregator.Report(
			healthName,
			&health.HealthReport{Live: true, Ready: d.doneFirstApply},
		)
	}
```
`d.config.HealthAggregator` should be OK because only set at start of day.  But maybe `d.doneFirstApply` could be changing in another goroutine, and we might need to delve further inside `d.config.HealthAggregator.Report`.

Reply:
Similarly the next 3 changes are all in goroutines.

Reply:
I think it's correct but I wish it was a bit simpler:

- `d.config.HealthAggregator` is start-of day config
- `doneFirstApply` only changes on the main goroutine and the goroutines from which I'm calling `reportHealth()` are short-lived: the main goroutine starts them and then waits for them to finish before continuing.
- `Report()` itself uses a mutex to protect the HealthAggregator.

Reply:
OK, thanks for the reassurance.